dev-docs/plugins-modules-packages.adoc

```diff
@@ -0,0 +1,129 @@
+= Plugins, Modules and Packages
+:toc: left
+
+This document discusses how developers can group features into Plugins, Modules and Packages.
+
+== Definitions
+
+=== Plugin
+
+A plugin is a low level class that implements a certain Solr interface, such as
+`QParserPlugin`, `AuthPlugin`, `UpdateRequestProcessor`, `QueryResponseWriter` etc.
+See ref guide page https://solr.apache.org/guide/solr-plugins.html[Solr Plugins] for details.
+
+=== Module
+
+Earlier referred to as "Contrib". This is a well-defined piece of functionality that is
```

Comment:
nit-pick: Prefer to start a description on what something *is*, and not what it *is not*.

Reply:
In no place here did you say a module is a set of plugins.  I mean... isn't that like the most basic description?

Reply:
Last sentence says "A module can contain multiple Plugins."
I'm open for re-phrasing suggestions.

Reply:
Re-phrased this, please see the update.